core/controller/src/main/scala/whisk/core/loadBalancer/LoadBalancerToKafka.scala

```diff
@@ -46,25 +46,22 @@ trait LoadBalancerToKafka extends Logging {
     /**
      * Publishes message on kafka bus for the invoker to pick up.
      *
-     * @param topic the topic name extracted from URI
-     * @param msg the message received via POST
-     * @param transid the transaction id, this may be the tid assigned by the controller and carried by the message or one determined by the load balancer service
-     * @return msg to return in HTTP response
+     * @param msg the activation message to publish on an invoker topic
+     * @param transid the transaction id for the request
+     * @return result of publishing the message as a Future
      */
-    def doPublish(component: String, msg: Message)(implicit transid: TransactionId): Future[Unit] = {
-        getTopic(component, msg) match {
+    def publish(msg: Message)(implicit transid: TransactionId): Future[Unit] = {
+        getInvoker(msg) map {
+            i => (i, Message.invoker(i))
```

Comment:
Why the indirection with a `Tuple2` and not just `getInvoker(msg) match ...` and build the topicname in the `Some` case?